src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/controller/AuthenticationController.java

```diff
@@ -0,0 +1,139 @@
+package ch.uzh.ifi.fs22.sel.group15.cozycave.server.controller;
+
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.entity.Location;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.entity.user.AuthenticationData;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.entity.user.User;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.entity.user.UserDetails;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.rest.dto.UserGetDto;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.rest.dto.UserPostPutDto;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.rest.mapper.UserMapper;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.security.JwtTokenProvider;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.service.UserService;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/v1/auth")
+// TODO: add logout possibility if required (individual token for each user)
+public class AuthenticationController {
+
+    private final Logger log = LoggerFactory.getLogger(AuthenticationController.class);
+
+    private final UserService userService;
+    private final PasswordEncoder passwordEncoder;
+    private final AuthenticationManager authenticationManager;
+    private final JwtTokenProvider jwtTokenProvider;
+
+    public AuthenticationController(UserService userService, PasswordEncoder passwordEncoder,
+        AuthenticationManager authenticationManager, JwtTokenProvider jwtTokenProvider) {
+        this.userService = userService;
+        this.passwordEncoder = passwordEncoder;
+        this.authenticationManager = authenticationManager;
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @PostMapping("/register")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ResponseBody
+    public UserGetDto register(@RequestBody UserPostPutDto userPostPutDto) {
+        log.debug("new user registration try: {}", userPostPutDto.toString());
+
+        Location address = null;
+        if (userPostPutDto.getDetails().getAddress() != null) {
+            address = new Location(
+                "home",
```

Comment:
shouldn't the user specify this through userPostPutDto.getDetails().getAddress().getName()? Because of "business" addresses

Reply:
should this then be required?

Reply:
yes I guess, but we can do this later as well.